x/oracle/keeper/native_token.go

```diff
@@ -0,0 +1,291 @@
+package keeper
+
+import (
+	"errors"
+	"fmt"
+
+	sdkmath "cosmossdk.io/math"
+	"github.com/ExocoreNetwork/exocore/x/oracle/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// deposit: update staker's totalDeposit
+// withdoraw: update staker's totalDeposit
+// delegate: update operator's price, operator's totalAmount, operator's totalShare, staker's share
+// undelegate: update operator's price, operator's totalAmount, operator's totalShare, staker's share
+// msg(refund or slash on beaconChain): update staker's price, operator's price
+
+const maxEffectiveBalance = 32
+
+var stakerList types.StakerList
+
+// GetStakerInfo returns details about staker for native-restaking under asset of assetID
+func (k Keeper) GetStakerInfo(ctx sdk.Context, assetID, stakerAddr string) types.StakerInfo {
+	store := ctx.KVStore(k.storeKey)
+	stakerInfo := types.StakerInfo{}
+	value := store.Get(types.NativeTokenStakerKey(assetID, stakerAddr))
+	if value == nil {
+		return stakerInfo
```

Comment:
Consider returning an additional boolean value in the **GetStakerInfo** and **GetStakerList** functions to indicate whether the data has been found

> func (k Keeper) GetStakerInfo(ctx sdk.Context, assetID, stakerAddr string) (types.StakerInfo, bool) {
    // ...
    if value == nil {
        return stakerInfo, false
    }
    k.cdc.MustUnmarshal(value, &stakerInfo)
    return stakerInfo, true
}

Reply:
when not found, it will return empty struct

Reply:
stakerList is a wrapper of slice, so could be check with length